### PR TITLE
Fix for Pixelated Shadows

### PR DESCRIPTION
**Pull request by ballisticfox**
Fix per pixel lighting and blocklights not respecting the pixelated shadows setting.

TODO:
- Accumulation causes blurriness on the pixels, may be fixable.

### CODE REVIEW

**Comment by ballisticfox** [comment]
Shimmering artifacts in the last commit have been fixed and the pixels are now properly locked to the grid.

**Comment by gri573** [comment]
Thank you for your pull request! This is more-or-less directly transferrable to the singlesample-softshadows-v2 branch, which is the current development branch. I will also add some comments where I would prefer things to be done slightly differently, but I can adapt that myself if you don't want to put more effort into this PR. And I'd like to keep indentation using spaces rather than tabs, as that is what we (emin (Complementary developer), spacEagle17 and isuewo (Euphoria Patches developers and myself) recently agreed upon.

File: shaders/program/prepare4_csh.glsl [hunk 1]
```diff
@@ -44,12 +44,14 @@ void main() {
...
     if (validData) {
-        vec4 playerPos = gbufferModelViewInverse * (gbufferProjectionInverse * (vec4((readTexelCoord + 0.5) / view, 1 - normalDepthData.a, 1) * 2 - 1));
-        playerPos /= playerPos.w;
-        #if PIXEL_SHADOW > 0 && !defined GBUFFERS_HAND
-			playerPos = floor((playerPos) * (PIXEL_SHADOW) + 0.001) / (PIXEL_SHADOW);
+		vec4 playerPos = gbufferModelViewInverse * (gbufferProjectionInverse * (vec4((readTexelCoord + 0.5) / view, 1 - normalDepthData.a, 1) * 2 - 1));
+		playerPos /= playerPos.w;
+		vec3 playerPosM = vec3(playerPos.x,playerPos.y,playerPos.z);
+    	#if PIXEL_SHADOW > 0 && !defined GBUFFERS_HAND
+			vxPos = floor(playerToVx(playerPosM) * (PIXEL_SHADOW) + 0.001) / (PIXEL_SHADOW) + max(0.1, 0.005 * length(playerPosM)) * normalDepthData.xyz;
```
[the remark this answers is not in the document]
Comment (ballisticfox) [reply shaders/program/prepare4_csh.glsl:51]: I'll investigate this after class today 👍

**Comment by ballisticfox** [comment]
I'm curious, should [lines 152-154](https://github.com/ballisticfox/rethinking-voxels/blob/588ced1ba9f5142b049ca9791f9ceffd78107635/shaders/lib/lighting/mainLighting.glsl#L152-L154) in mainLighting.glsl also have the `!defined GBUFFERS_HAND` preprocessor instruction?
```
#if PIXEL_SHADOW > 0 && !defined GBUFFERS_HAND
    playerPosM = floor((playerPosM + cameraPosition) * PIXEL_SHADOW + 0.001) / PIXEL_SHADOW - cameraPosition + 0.5 / PIXEL_SHADOW;
#endif
```

**Comment by gri573** [comment]
> I'm curious, should [lines 152-154](https://github.com/ballisticfox/rethinking-voxels/blob/588ced1ba9f5142b049ca9791f9ceffd78107635/shaders/lib/lighting/mainLighting.glsl#L152-L154) in mainLighting.glsl also have the `!defined GBUFFERS_HAND` preprocessor instruction?
> 
> ```glsl
> #if PIXEL_SHADOW > 0 && !defined GBUFFERS_HAND
>     playerPosM = floor((playerPosM + cameraPosition) * PIXEL_SHADOW + 0.001) / PIXEL_SHADOW - cameraPosition + 0.5 / PIXEL_SHADOW;
> #endif
> ```

That one should have it, as that file may be included in the gbuffers_hand program, and the pixelated shadows shouldn't appear on the player's hand, as they look weird there.

File: shaders/program/prepare4_csh.glsl (hunk 1 above)
[the remark this answers is not in the document]
Comment (ballisticfox) [reply shaders/program/prepare4_csh.glsl]: I added the change as requested, is there any specific test case for this?

**Comment by gri573** [comment]
Thank you

**Pull request merged**